src/contracts/registry.ts

```diff
@@ -0,0 +1,68 @@
+import { getContractAddress } from "./contract";
+import { HexString } from "../types/Strings";
+import { getConfig } from "../config/config";
+import { ContractTransaction } from "ethers";
+import { MissingContract, MissingProvider, MissingSigner } from "../utilities";
+import { Registry__factory } from "../types/typechain";
+
+const CONTRACT_NAME = "Registry";
+
+/**
+ * Get the JSON RPC error from the body, if one exists
+ * @param e The error expected to have a vm Error
+ *
+ * @returns the error if any
+ */
+const getVmError = (e: { body?: string }): string | undefined => {
+  try {
+    const parsed = JSON.parse(e.body || "{}");
+    return parsed?.error?.message;
+  } catch (e) {
+    return undefined;
```

Comment:
how can this be reached?

Reply:
If the error is not a VM error it will fail to parse.